manifests/init.pp

```diff
@@ -53,7 +53,9 @@
 # $authentication::               Enable user authentication. Initial credentials are set using admin_username
 #                                 and admin_password.
 #
-# $passenger::                    Configure foreman via apache and passenger
+# $apache::                       Configure Foreman via Apache. By default via passenger but otherwise as a reverse proxy.
+#
+# $passenger::                    Whether to configure Apache with passenger or as a reverse proxy.
```

Comment:
Should these two conflict? or is the idea you need both now to get passenger?

Reply:
This changes it so that the parameter is ignored if `apache` is `false`

Reply:
A future-proof approach, if we consider multiple kind of setups, could be an `Enum['apache+passenger', 'apache+puma', 'puma', 'nginx+passenger', 'nginx+puma'], what do you think about?

Reply:
While I like the idea of supporting nginx, I think it would be premature. Longer term I'd like to deprecate passenger. It'd be better to decide then how to best support it.

Reply:
I agreed about nginx, I noted it as example. The idea I want to point is to have a single class variable to select the targeted mode. Internally, we could keep `$passenger`, `$apache` and `$use_foreman_service`.